### PR TITLE
FlatArray

### PR DESCRIPTION
**Pull request by priyang12**
Created FlatArray Function which is a flat function for flatting array using recursion. i have added tests to validate the function with depths.

### CODE REVIEW

File: JS/recursion/FlatArray.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/* 
+    Flat Array using Recursion
+
+    Ex - Full Flat
+    Input: [0,[1,2,3,[4,5]],6,[7,[8,9]]]
+    OutPut: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+
+*/
+
+export const FlatArray = (InputArr, Depth = Infinity, CurrentDepth = 0) => {
```
Comment (anarchymonkey) [inline JS/recursion/FlatArray.js]: method names and variable names should be `camelCase` as per industry standards

File: JS/recursion/FlatArray.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  let flattened = [];
+  const GlobalDepth = Depth;
+  for (let i = 0; i < InputArr.length; i++) {
+    const element = InputArr[i];
```
Comment (anarchymonkey) [inline JS/recursion/FlatArray.js]: Pro tip: 
```
inputArr.at(i);
```

File: JS/recursion/FlatArray.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    if (Array.isArray(element) && element.length > 0) {
+      if (GlobalDepth === CurrentDepth) {
+        flattened.push(element);
+      } else {
+        const Flatten = FlatArray(element, GlobalDepth, CurrentDepth + 1);
```
Comment (anarchymonkey) [inline JS/recursion/FlatArray.js]: let's change the methodName from FlatArray to `flattenArray` as method names should be verbs not nouns.

**Review by anarchymonkey — CHANGES_REQUESTED** [review]
Loved the PR, added some comments. 

Thanks for contributing!

**Comment by priyang12** [comment]
@anarchymonkey hi, I have made the changes.

**Review by anarchymonkey — APPROVED** [review]

**Pull request merged**